### PR TITLE
Reset CallTraceStorage counters before reporting live objects.

### PR DESCRIPTION
**Pull request by krk**
Allows collecting allocation and live object traces at the same time. 
Also fixes "sometimes missing" stack traces in allocations.

### Related issues
https://github.com/async-profiler/async-profiler/issues/928

### Motivation and context
Enabling live object tracing missed stack traces "sometimes".

### How has this been tested?
Unit tests asserting on live object's size with varying GC probability. 

For the stack traces, JFR recording is created and allocation samples are verified to have stack traces in the JFR.

Sample test assertion *without* the patch, that is observed "sometimes":

```
INFO: Running AllocTests.livenessJfr...
WARNING: AllocTests.livenessJfr failed
java.lang.AssertionError: Stack trace missing for id 4939
        at test.alloc.AllocTests.livenessJfr(AllocTests.java:139)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at one.profiler.test.Runner.run(Runner.java:152)
        at one.profiler.test.Runner.run(Runner.java:168)
        at one.profiler.test.Runner.main(Runner.java:217)
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: src/callTraceStorage.cpp [hunk 1]
```diff
@@ -289,3 +289,18 @@ void CallTraceStorage::add(u32 call_trace_id, u64 counter) {
...
+void CallTraceStorage::resetCounters() {
+     for (LongHashTable* table = _current_table; table != NULL; table = table->prev()) {
+        u64* keys = table->keys();
+        CallTraceSample* values = table->values();
+        u32 capacity = table->capacity();
+
+        for (u32 slot = 0; slot < capacity; slot++) {
+            if (keys[slot] != 0) {
+                CallTraceSample& s = values[slot];
+                storeRelease(s.counter, 0);
```
Comment (apangin) [inline src/callTraceStorage.cpp:302]: Don't we need to reset `s.samples` too?
Check that flame graph is generated correctly both with and without `--total` option.

File: test/test/alloc/AllocTests.java [hunk 2]
```diff
@@ -55,4 +61,69 @@ public void humongous(TestProcess p) throws Exception {
...
+    @Test(mainClass = RandomBlockRetainer.class, args = "0.9", inputs = { "true" })
+    @Test(mainClass = RandomBlockRetainer.class, args = "1.0", inputs = { "true" })
+    public void liveness(TestProcess p) throws Exception {
+        final long TOTAL_BYTES = 50000000;
+        final double tolerance = 0.10;
+
+        final boolean live = Boolean.parseBoolean(p.inputs()[0]);
+        final double keepChance = live ? Double.parseDouble(p.test().args()) : 1.0;
+
+        Output out = p.profile("--alloc 1k --total -o collapsed" + (live ? " --live" : ""));
```
Comment (apangin) [inline test/test/alloc/AllocTests.java:87]: `--live` is supported on JDK 11+ only. The does should not run or should fail on JDK 8.

File: src/objectSampler.cpp [hunk 3]
```diff
@@ -100,6 +100,10 @@ class LiveRefs {
         jvmtiEnv* jvmti = VM::jvmti();
         Profiler* profiler = Profiler::instance();
 
+        // Set CallTraceStorage counters to zero, so only live objects
+        // will be reported to collapsed and flamegraph outputs.
+        profiler->resetCounters();
```
Comment (apangin) [inline src/objectSampler.cpp:105]: The idea is to reset counters only for non-JFR recording.
In JFR recording, each sample is recorded individually, so accumulated counters are not actually used.

File: test/one/profiler/test/Assert.java [hunk 4]
```diff
@@ -12,6 +12,9 @@ public class Assert {
     private static final Logger log = Logger.getLogger(Assert.class.getName());
 
     public static void isGreater(double value, double threshold) {
+        if (value <= threshold) {
+            throw new AssertionError("Expected " + value + " > " + threshold);
+        }
```
Comment (apangin) [inline test/one/profiler/test/Assert.java:17]: Redundant lines after merge?

Comment (krk) [reply test/one/profiler/test/Assert.java]: Good catch, yes!

File: test/one/profiler/test/Assert.java [hunk 5]
```diff
@@ -25,6 +25,12 @@ public static void isGreater(double value, double threshold, String message) {
         }
     }
 
+    public static void isGreaterOrEqual(double value, double threshold) {
+        if (value < threshold) {
+            throw new AssertionError("Expected " + value + " >= " + threshold);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (krk) [reply test/one/profiler/test/Assert.java:32]: These are used in the current commit, happy to replace them with #1027 when it is merged.

**Pull request merged**